app/metadata.js

```diff
@@ -1,79 +1,109 @@
-const statsdClient = require('./statsd');
 const domino = require('domino');
+const fastimage = require('fastimage');
+const statsdClient = require('./statsd');
 const {getMetadata, makeUrlAbsolute} = require('page-metadata-parser');
 require('isomorphic-fetch');
 
-function getDocumentMetadata(url, window) {
-  const doc = window.document;
-  const metadata = getMetadata(doc, url);
+// See https://github.com/mozilla/page-metadata-service/issues/89#issuecomment-243931889
+fastimage.threshold(-1);
+
+function fetchUrlContent(url) {
+  const startFetch = statsdClient.getTimestamp();
+
+  return fetch(url).then(res => {
+    const endFetch = statsdClient.getTimestamp();
+    statsdClient.timing('fetch_time', (endFetch - startFetch));
+
+    if (res.status >= 200 && res.status < 300) {
+      statsdClient.increment('fetch_success');
+      return res;
+    } else {
+      throw new Error(`Request Failure: ${res.status} ${res.statusText}`);
+    }
+  }).then(res => res.text())
+  .catch(e => {
+    statsdClient.increment('fetch_fail');
+
+    const endFetch = statsdClient.getTimestamp();
+    statsdClient.timing('fetch_time', (endFetch - startFetch));
+
+    throw e;
+  });
+}
+
+function getHtmlDocument(html) {
+  return domino.createWindow(html).document;
+}
+
+function getImageInfo(imageUrl) {
+  const startImage = statsdClient.getTimestamp();
 
-  const responseData = {
-    url: metadata.url,
+  return fastimage.info(imageUrl).then(info => {
+    const endImage = statsdClient.getTimestamp();
+    statsdClient.timing('image_time', (endImage - startImage));
+    statsdClient.increment('image_success');
+
+    return info;
+  }).catch(e => {
+    const endImage = statsdClient.getTimestamp();
+    statsdClient.timing('image_time', (endImage - startImage));
+    statsdClient.increment('image_fail');
+
+    throw e;
+  });
+}
+
+function getDocumentMetadata(url, doc) {
+  const parsedMetadata = getMetadata(doc, url);
+
+  const urlMetadata = {
+    url: parsedMetadata.url,
     original_url: url,
```

Comment:
original_url is here to maintain backwards compatibility with the API defined by Embedly, and is completely unnecessary in the parser, since everyone that invokes the parser will already have the original url when they call.